src/rust/engine/process_execution/bazel_protos/src/metadata.rs

```diff
@@ -1,16 +1,74 @@
-use protobuf::Message;
 use std::collections::BTreeMap;
+use std::sync::Arc;
+
+use cpython::{ObjectProtocol, PyObject, Python};
+use parking_lot::Mutex;
+use protobuf::Message;
+
+#[derive(Clone)]
+pub enum RequestHeaders {
```

Comment:
This should probably be moved out of `bazel_protos` so there is no dependence on the `cpython` crate. Any thoughts on where to put this? (Same logic applies to `call_option` to.)

Reply:
I think slightly cleaner could be if the `Dynamic` took an `impl Fn` rather than a `PyObject`, and the thing which constructs it in `engine` could wrap the `cpython` into a regular Rust closure. This also makes testing a bunch easier, by not having the `cpython` dep here.

But if you want to keep this being a `PyObject`, a new crate (`rpc_utils`), or if you really want to avoid that, `store` (which is already depended on by `process_execution`)

Reply:
I originally used an `Box<dyn Fn ...>` here but there were issues with what it was capturing and borrow checker horribleness so I gave up and passed in the `PyObject` instead. I can retry the earlier iteration but with a trait and just pass in a `Box<dyn Trait>)` (with appropriate Arc/Mutex usage). That might side step the issues with having used a closure capturing who knows what that could not be cloned.

Reply:
I'd be happy to help with that if you need. Can ping in #slack.

Reply:
I suspect `Arc<Mutex<FnMut() -> BTreeMap<String, String>>>` is what you're looking for (or possibly `Arc<Fn() -> BTreeMap<String, String>>` - also happy to help out if needed!